http_rules/test_http_tables.py

```diff
@@ -0,0 +1,302 @@
+        hdr_uagent == "Mozilla*" -> chain2;
+        hdr_ref == "*.com" -> chain1;
+        hdr_ref == "http://example.*" -> chain3;
+        hdr_host == "bad.host.com" -> block;
+        hdr_host == "bar*" -> vh5;
+        mark == 1 -> vh7;
+        mark == 2 -> vh6;
+        mark == 3 -> vh5;
+        mark == 4 -> vh4;
+        mark == 5 -> vh3;
+        mark == 6 -> vh2;
+        mark == 7 -> vh1;
+        }
+        """
+    }
+
+    clients = [
+        {
+            'id' : 'gen',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80'
+        }
+    ]
+
+    requests_opt = [
+        (
+            ('/static/index.html'),
+            ('referer'),
+            ('example.com'),
+            False
+        ),
+        (
+            ('/script.php'),
+            ('referer'),
+            ('http://example.com/cgi-bin/show.pl'),
+            False
+        ),
+        (
+            ('/foo/example.com'),
+            ('host'),
+            ('test.app.com'),
+            False
+        ),
+        (
+            ('/bar/index.html'),
+            ('user-agent'),
+            ('Mozilla/60.0'),
+            False
+        ),
+        (
+            ('/static/foo/app/test.php'),
+            ('host'),
+            ('bar.example.com'),
+            False
+        ),
+        (
+            ('/app/hacked.com'),
+            ('referer'),
+            ('http://example.com'),
+            True
+        ),
+        (
+            ('/'),
+            ('host'),
+            ('bad.host.com'),
+            True
+        )
+    ]
+
+    chains = []
+    match_rules_test = True
+
+    def init_chain(self, (uri, header, value, block)):
+        ch = chains.base(uri=uri)
+        if block and self.match_rules_test:
+            ch.fwd_request = None
+        else:
+            for request in [ch.request, ch.fwd_request]:
+                request.headers.delete_all(header)
+                request.headers.add(header, value)
+                request.update()
+        self.chains.append(ch)
+
+    def setUp(self):
+        del(self.chains[:])
+        count = len(self.requests_opt)
+        for i in range(count):
+            self.init_chain(self.requests_opt[i])
+        tester.TempestaTest.setUp(self)
+
+    def wait_all_connections(self, tmt=1):
+        sids = self.get_servers_id()
+        for id in sids:
+            srv = self.get_server(id)
+            if not srv.wait_for_connections(timeout=tmt):
+                return False
+        return True
+
+    def start_all(self):
+        self.start_all_servers()
+        self.start_tempesta()
+        self.assertTrue(self.wait_all_connections())
+        self.get_client('gen').start()
+
+    def process(self, client, server, chain):
+        client.make_request(chain.request.msg)
+        client.wait_for_response(timeout=5)
+        if chain.fwd_request:
+            chain.fwd_request.set_expected()
+        self.assertEqual(server.last_request, chain.fwd_request)
```

Comment:
Not all assertions are processed: if block action is hit (and `chain.fwd_request is None`), then need to assert that the client connection was closed by Tempesta.

Reply:
Done.